getting-started/troubleshooting.html.md

```diff
@@ -1,176 +1,223 @@
+
+// ESM
+import Fastify from 'fastify'
+const fastify = Fastify({
+  logger: true
+})
+// CommonJs
+const fastify = require('fastify')({
+  logger: true
+})
 
 // Declare a route
-fastify.get('/', async (request, reply) => {
-  return { hello: 'world' }
+fastify.get('/', function (request, reply) {
+  reply.send({ hello: 'world' })
 })
 
 // Run the server!
-const start = async () => {
-  try {
-    await fastify.listen({ port: 8080 })
-    fastify.log.info(`server listening on ${fastify.server.address().port}`)
-  } catch (err) {
+fastify.listen({ port: 3000 }, function (err, address) {
+  if (err) {
     fastify.log.error(err)
     process.exit(1)
   }
-}
-start()
+  // Server is now listening on ${address}
+})
 ```
 
-If we build this locally and run it, it works just fine. Deploy it to fly though and we get:
+This example will work locally, but when you run `fly deploy` you’ll get the “app is listening on the incorrect address” warning.
 
-```
-Monitoring Deployment
-You can detach the terminal anytime without stopping the deployment
-
-1 desired, 1 placed, 0 healthy, 1 unhealthy
-v1 failed - Failed due to unhealthy allocations - no stable job version to auto revert to
-***v1 failed - Failed due to unhealthy allocations - no stable job version to auto revert to
-```
-
-That's not good, and there's no obvious reference to a host there. Lucky for us the fastify listen function accepts an object for defining additional options like host. We can pass in "0.0.0.0" as the host to successfully run inside a Docker container like so:
+You can modify the example to listen on host `0.0.0.0` and to print a log with the listening address:
 
 ```jsx
 ...
-const start = async () => {
-  try {
-    await fastify.listen({ port: 8080, host: "0.0.0.0" })
-...
+
+fastify.listen({ port: 3000, host: '0.0.0.0' }, function (err, address) {
+  if (err) {
+    fastify.log.error(err)
+    process.exit(1)
+  }
+  fastify.log.info(`server listening on ${address}`)
+})
 ```
 
-And redeploy that, it will work.
+Then make sure that the `internal_port` value in `fly.toml` is set to `3000`. 
 
-## _Inspecting with SSH_
+## Health checks failing
 
-You can use `flyctl ssh console` to connect to a running instance of your application. Use `flyctl ssh console -s` to select a specific instance.
+We don't do health checks when you launch your app for the first time, even if you use a pre-configured `fly.toml` file with health checks. But on subsequent deploys, health checks can fail for a number of reasons.
 
-## _Health checks failing_
+A good first step can be to look at a failed Machine and see what you can figure out. To see the specific Machine status, run `fly status --all` to get a list of Machines in your app. Find one with state `failed` or `stopped`, then run `fly machine status <machine id>` . This will give you a lot more information. Make sure you check the exit code, if it’s 0 it means health check failures, if it’s not zero it’s some issue crashing the process.
```

Comment:
```suggestion
A good first step can be to look at a failed Machine and see what you can figure out. To see the specific Machine status, run `fly status --all` to get a list of Machines in your app. Find one with state `failed` or `stopped`, then run `fly machine status <machine id>` . This will give you a lot more information. Make sure you check the exit code: if it’s non-zero, your process crashed.
```

If a `stopped` machine has exit code 0, that could be a deliberate `fly m stop`, or an intentional process exit. Right? Also, do we actually get `failed` in `fly status`? Is that something a health check does? Wait, do health checks stop Machines these days? I haven't been adding health checks to my apps and since they don't get them by default, I'm not familiar with the behaviour anymore.

Reply:
Yes, health checks are a whole other topic. For now, a quick fix is to remove the bit about "stopped or failed" state. They can still look at Machine status. Will revisit this.